src/coreclr/debug/daccess/dacdbiimpl.cpp

```diff
@@ -2315,6 +2315,10 @@ void DacDbiInterfaceImpl::GetClassTypeInfo(TypeHandle                      typeH
                                            DebuggerIPCE_ExpandedTypeData * pTypeInfo,
                                            AppDomain *                     pAppDomain)
 {
+    if (typeHandle.AsMethodTable()->IsContinuation())
```

Comment:
I would extract this conversion into a helper function such as:
```
typeHandle = UpCastTypeIfNeeded(typeHandle);
```

We might add other types not backed by metadata in the future and consolidating the conversion in a helper will let us do it consistently everywhere it needs to occur. The helper also provides a natural place to add comments describing why we want to do this conversion.